### ISSUE TITLE
[BUG][code-analyzer] Cannot Handle Commas in File or Folder Paths

### ISSUE THREAD

**Issue by vikashmangalfexle** [forcedotcom/code-analyzer#1949]
### Have you tried to resolve this issue yourself first?

- [x] I confirm I have gone through the above steps and still have an issue to report.

### Bug Description

When running Salesforce Code Analyzer (PMD scan) using the CLI, the command fails whenever the file or folder path contains a comma ( , ).
Even when the path is fully quoted, the CLI incorrectly splits the path at the comma and treats it as a separate argument, causing the scan to fail with a “file not found” error.

### Output / Logs

```shell
C:\Users\vikash.mangal>sf code-analyzer run --workspace "D:/New ,folder (2)/MyClass.cls"  --output-file .\pmd-result.html --rule-selector pmd:Security

Streaming logs in real time to:
    C:\Users\VIKASH~1.MAN\AppData\Local\Temp\sfca-2025_12_05_16_07_42_684.log

Error (1): The file or folder 'D:\New' does not exist.


Warning: The input format for array arguments has changed. Use this format: --array-flag value1 --array-flag value2 --array-flag value3
```

### Steps To Reproduce

1. Create a folder whose name contains a comma (e.g., "New ,folder (2)").
2. Place any text or Apex file inside the folder.
3. Run the following command:
`sf code-analyzer run --workspace "D:/New ,folder (2)/MyClass.cls"  --output-file .\pmd-result.html --rule-selector pmd:Security
`

Observe that Salesforce Code Analyzer fails to resolve the correct file path and throws a “file does not exist” error.

### Expected Behavior

- The CLI should correctly parse file paths containing commas, especially when the path is enclosed in double quotes.
- The analyzer should treat the entire quoted path as a single argument and process it without splitting at the comma.
- PMD scan should run successfully for paths with commas, just like it does for paths containing spaces or parentheses.

### Operating System

Windows 11

### Salesforce CLI Version

@salesforce/cli/2.114.5 win32-x64 node-v22.21.1

### Code Analyzer Plugin (code-analyzer) Version

code-analyzer 5.7.0

### Node Version

node-v22.21.1

### Java Version

openjdk version "21.0.5" 2024-10-15 LTS

### Python Version

Python 3.11.9

### Additional Context (Screenshots, Files, etc)

<img width="1891" height="378" alt="Image" src="https://github.com/user-attachments/assets/b0543683-e9ea-4f67-97ed-95ab2d571152" />

### Urgency

Moderate
